docs/reference/sdk-guidelines.mdx

```diff
@@ -0,0 +1,678 @@
+```
+Result output: [CollectionObject](https://docs.onflow.org/fcl/reference/api/#collectionobject)
+
+### Execute Scripts
+[<img src="https://raw.githubusercontent.com/onflow/sdks/main/templates/documentation/ref.svg" width="130">](https://docs.onflow.org/fcl/reference/api/#query)
+
+Scripts allow you to write arbitrary non-mutating Cadence code on the Flow blockchain and return data. You can learn more about [Cadence and scripts here](https://docs.onflow.org/cadence/language/), but we are now only interested in executing the script code and getting back the data.
+
+We can execute a script using the latest state of the Flow blockchain or we can choose to execute the script at a specific time in history defined by a block height or block ID.
+
+📖 **Block ID** is SHA3-256 hash of the entire block payload, but you can get that value from the block response properties.
+
+📖 **Block height** expresses the height of the block in the chain.
+
+```javascript
+import * as fcl from "@onflow/fcl";
+
+const result = await fcl.query({
+  cadence: `
+    pub fun main(a: Int, b: Int, addr: Address): Int {
+      log(addr)
+      return a + b
+    }
+  `,
+  args: (arg, t) => [
+    arg(7, t.Int), // a: Int
+    arg(6, t.Int), // b: Int
+    arg("0xba1132bc08f82fe2", t.Address), // addr: Address
+  ],
+});
+```
+Example output:
+```bash
+console.log(result); // 13
+```
+
+## Mutate Flow Network
+Flow, like most blockchains, allows anybody to submit a transaction that mutates the shared global chain state. A transaction is an object that holds a payload, which describes the state mutation, and one or more authorizations that permit the transaction to mutate the state owned by specific accounts.
+
+Transaction data is composed and signed with help of the SDK. The signed payload of transaction then gets submitted to the access node API. If a transaction is invalid or the correct number of authorizing signatures are not provided, it gets rejected. 
+
+## Transactions
+A transaction is nothing more than a signed set of data that includes script code which are instructions on how to mutate the network state and properties that define and limit it's execution. All these properties are explained bellow. 
+
+📖 **Script** field is the portion of the transaction that describes the state mutation logic. On Flow, transaction logic is written in [Cadence](https://docs.onflow.org/cadence/). Here is an example transaction script:
+```
+transaction(greeting: String) {
+  execute {
+    log(greeting.concat(", World!"))
+  }
+}
+```
+
+📖 **Arguments**. A transaction can accept zero or more arguments that are passed into the Cadence script. The arguments on the transaction must match the number and order declared in the Cadence script. Sample script from above accepts a single `String` argument.
+
+📖 **[Proposal key](https://docs.onflow.org/concepts/transaction-signing/#proposal-key)** must be provided to act as a sequence number and prevent reply and other potential attacks.
```

Comment:
prevent REPLAY and other potential attacks.

Reply:
good catch